src/hdl_graph_slam/map_saver.cpp

```diff
@@ -0,0 +1,38 @@
+#include "ros/ros.h"
+#include <hdl_graph_slam/SaveMap.h>
+#include <string.h>
+#include <signal.h>
+
+ros::ServiceClient client;
+hdl_graph_slam::SaveMap savemap;
+
+void save_interrupt(int s) {
+    // Save map
+    client.call(savemap);
+
+    // Restore default behaviour of SIGINT
+    signal(SIGINT, SIG_DFL);
+}
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "map_saver");
+    ros::NodeHandle n;
+
+    // Form message
+    float res = 0.05;
+    std::string des = argv[1];
```

Comment:
read these values from a rosparam.

Reply:
res as a rosparam sounds good, but I think destination should be a cmd line arg to be consistent with the other SLAMs. Actually why not just make res a cmd line arg as well with default 0.05, then expose as an arg in process.launch. e.g. `roslaunch 3d_Slam_tools process algo:=hdl res:=0.1 ...`